recipes/network_interfaces_config.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: aws-parallelcluster
+# Recipe:: network_interfaces_config
+#
+# Copyright 2013-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+def network_interface_macs
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs")
+  res = Net::HTTP.get_response(uri)
+  macs = res.body.gsub("/", "").split("\n")
+  macs
+end
```

Comment:
As per our conversation:
- Looks like it's simpler using the metadata url because it returns just the list of macs.
- node['macaddress'] returns the mac of the primary NIC